dig_test.go

```diff
@@ -539,15 +524,19 @@ func TestCantProvideParameterObjects(t *testing.T) {
 		require.Contains(t, err.Error(), "can't provide parameter objects")
 	})
 
-	t.Run("constructor pointer", func(t *testing.T) {
+	t.Run("pointer from constructor", func(t *testing.T) {
 		type Args struct{ Param }
 
+		args := &Args{}
+
 		c := New()
-		err := c.Provide(func() (*Args, error) {
-			panic("great sadness")
-		})
-		require.Error(t, err, "provide should fail")
-		require.Contains(t, err.Error(), "can't provide parameter objects")
+		require.NoError(t, c.Provide(func() (*Args, error) {
+			return args, nil
+		}), "provide failed")
+		require.NoError(t, c.Invoke(
+			func(a *Args) {
+				require.True(t, args == a, "args must match")
+			}), "invoke failed")
```

Comment:
I try to start anonymous functions on new lines but I see that the rest of the code doesn't do that. Fixing.